vscode-extension/package.json

```diff
@@ -52,7 +52,12 @@
         "metals.sbt.enabled": {
           "type": "boolean",
           "default": false,
-          "description": "EXPERIMENTAL. Use sbt language server to compile on file save and report diagnostics. Requires sbt 1.1.0"
+          "description": "EXPERIMENTAL. Use sbt language server to compile on file save and report diagnostics. Requires sbt 1.1+."
```

Comment:
s/and report diagnostics//

In what situation would you want this enabled when `command=""` and `diagnostics=disabled`?

Reply:
That's probably not a realistic scenario. An alternative approach would be to derive the `sbt.enabled` by the other settings i.e. when `command=""` and `diagnostic.enabled=false` then don't try to connect to sbt server.

That said, I'm not super into this solution, since it seems we would be mixing concerns.

Reply:
We can go with this approach if we want to enable sbt integration by default, because `sbt.command` shouldn't be empty by default. 

Currently user just ticks one checkbox (`sbt.enabled`) and gets expected behaviour: compile on save and get diagnostics. If we set `sbt.diagnostics.enabled = false` + `sbt.command = ""`, then user will have to enter a command (which is not absolutely obvious IMO) and then still won't get diagnostics, because they didn't check the `sbt.diagnostics` setting. I think this can be easily confusing. So in the end I vote for leaving `sbt.enabled` setting.

Reply:
> don't try to automatically connect to sbt server, user must trigger "Connect to sbt server" command to get error message about connection problems. We might try to connect in the background if there is a project/target/active.json and silently fail if there is no sbt server.

This is https://github.com/scalameta/metals/issues/253.

I'm fine with the proposed solution, but I'd like to minimize changes in this PR and leave the behaviour described in #253 for another PR:

* set `sbt.command = ""` and `sbt.diagnostics.enabled = true`
* remove `sbt.enabled`
* remove auto-connection

would this be enough?

Reply:
Those proposed changes sound good. The smart detection can be delegated to #253

Reply:
Done in 2f1ad16. And I think it works as expected:
* doesn't auto-connect
* if you connect, but have empty `sbt.command`, nothing happens on save
* if you set `sbt.command`, it runs it on save
* if you don't set it, but run manually, say `~metals/compile`, it will broadcast diagnostics (and you will receive them in the client if `sbt.diagnostics` is enabled)

I like it more than before: it feels like you get more control and it doesn't annoy you with the connection error when you forgot to launch sbt before opening the project. But #253 will improve it even more.